lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -128,7 +165,76 @@ impl MessageState {
     }
 }
 
-#[derive(Debug, Snafu, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Eq)]
+pub enum ChunkedGelfDecompression {
+    Gzip,
+    Zlib,
+    None,
+}
+
+impl ChunkedGelfDecompression {
+    // TODO: add tests for compression detection
+    pub fn from_magic(data: &Bytes) -> Self {
+        if data.starts_with(GZIP_MAGIC) {
+            debug!("Detected Gzip compression");
+            return Self::Gzip;
+        }
+
+        if data.starts_with(ZLIB_MAGIC) {
+            if let Some([first_byte, second_byte]) = data.get(0..2) {
+                // TODO: add inspired from https://github.com/grafana/go-gelf/blob/25db8704bcf3f484c958312cd0cc49e5c768dcf1/gelf/reader.go#L120
+                // and https://github.com/Graylog2/graylog2-server/blob/1bdeca5f8ad3d31ff0c1c7981978ae8c4d300ebc/graylog2-server/src/main/java/org/graylog2/inputs/codecs/gelf/GELFMessage.java#L157
+                if (*first_byte as u16 * 256 + *second_byte as u16) % 31 == 0 {
+                    debug!("Detected Zlib compression");
+                    return Self::Zlib;
+                }
+            };
```

Comment:
Should we return an error if the `if let Some([first_byte, second_byte]) = data.get(0..2)` or `if (*first_byte as u16 * 256 + *second_byte as u16) % 31 == 0` branches does not succeed? For example, any data that starts with `ZLIB_MAGIC` but does not satisfy `(*first_byte as u16 * 256 + *second_byte as u16) % 31 == 0`

Right now, we return `None` compression, but Gelf Server implementation returns an error https://github.com/Graylog2/graylog2-server/blob/1bdeca5f8ad3d31ff0c1c7981978ae8c4d300ebc/graylog2-server/src/main/java/org/graylog2/inputs/codecs/gelf/GELFMessage.java#L160

Reply:
We could add a `warn!` here (and return `None`) since I can imagine this being hard to debug without any hint.

Reply:
https://github.com/vectordotdev/vector/pull/21816/commits/c0bfabcfc8b5f1bbe37dc494b941b65ea4691567